packages/vector_graphics/lib/src/listener.dart

```diff
@@ -261,6 +311,51 @@ class FlutterVectorGraphicsListener extends VectorGraphicsCodecListener {
     _canvas.clipPath(_paths[pathId]);
   }
 
+  @override
+  void onPatternStart(int patternId, double x, double y, double width,
+      double height, Float64List transform) {
+    assert(_currentPattern == null);
+    _currentPattern = PatternConfig(patternId, width, height, transform);
+    _patternRecorders[patternId] = ui.PictureRecorder();
+    final ui.Canvas newCanvas = ui.Canvas(_patternRecorders[patternId]!);
+    newCanvas.clipRect(ui.Offset(x, y) & ui.Size(width, height));
+    _patternCanvases[patternId] = newCanvas;
+  }
+
+  /// Creates ImageShader for active pattern.
+  Future<void> onPatternFinished(PatternConfig? currentPattern,
+      ui.PictureRecorder? patternRecorder, ui.Canvas canvas) async {
+    final FlutterVectorGraphicsListener patternListener =
+        FlutterVectorGraphicsListener._(
+            canvas, patternRecorder!, _locale, _textDirection);
+    double imageWidth = currentPattern!.width;
+    double imageHeight = currentPattern.height;
+
+    if (imageWidth < 1) {
+      imageWidth = (_size.width * imageWidth);
+    }
+    if (imageHeight < 1) {
+      imageHeight = (_size.width * imageHeight);
+    }
```

Comment:
Can you explain this a bit? Are you treating width/height as a % conditional on its value? Would be worth documenting this elsewhere too. I assume its px otherwise?

Reply:
I  actually realized I addressed percentage dimensions in parser.dart so I'll remove this.